consensus/babe/babe.go

```diff
@@ -17,48 +17,52 @@
 package babe
 
 import (
-	"crypto/rand"
 	"encoding/binary"
 	"errors"
 	"fmt"
 	"math"
 	"math/big"
 	"time"
 
+	schnorrkel "github.com/ChainSafe/go-schnorrkel"
 	"github.com/ChainSafe/gossamer/codec"
 	tx "github.com/ChainSafe/gossamer/common/transaction"
 	"github.com/ChainSafe/gossamer/core/types"
-	"github.com/ChainSafe/gossamer/keystore"
+	"github.com/ChainSafe/gossamer/crypto/sr25519"
 	"github.com/ChainSafe/gossamer/runtime"
 	log "github.com/ChainSafe/log15"
 )
 
-// Session contains the VRF keys for the validator
+// Session contains the VRF keys for the validator, as well as BABE configuation data
 type Session struct {
-	keystore       *keystore.Keystore
+	keypair        *sr25519.Keypair
 	rt             *runtime.Runtime
 	config         *BabeConfiguration
 	authorityIndex uint64
 	authorityData  []AuthorityData
 	epochThreshold *big.Int // validator threshold for this epoch
 	txQueue        *tx.PriorityQueue
-	isProducer     map[uint64]bool    // whether we are a block producer at a slot
+	isProducer     map[uint64][]byte  // whether we are a block producer at a slot
```

Comment:
Why `[]byte` instead of `bool` here?

Reply:
we want to store the proof that we're a block producer if we are, so I've changed it to store the proof in this mapping (the byte array will be nil if we're not a producer)

Reply:
That makes sense. We should probably update the comment and maybe the name as well.

Reply:
I will update the comment, not sure what would be a good name, like `slotToProof` or something?

Reply:
Yea, that works. `isProducer` implies a bool so keeping it could be a bit confusing but not totally necessary I guess.

Reply:
ok, renamed